template-only-docs/template-technical-design-principles.md

```diff
@@ -0,0 +1,48 @@
+# Template technical design principles
+
+This document outlines the technical design principles that the template-infra project follows. These principles guide the development of the infrastructure template.
+
+## Design for simplicity
+
+### Design for the common case
+
+- Optimize for the common case
+- Avoid adding unnecessary flexibility that increases complexity
+- Ensure that uncommon cases can be addressed through explicit customization
+
+### Foster developer understanding and ownership
+
+- Ensure the infrastructure is transparent and comprehensible to project teams.
+- Encourage developers to understand the inner workings of the infrastructure so they are equipped to extend or customize it as needed.
+
+### Maintain a level of complexity comparable to what project teams would manage if building the functionality themselves
+
+- Avoid adding complexity to the project that arises from the fact that the code came from a template
+- Avoid adding complexity to the project that arises from the need to support multiple use cases
+
+### Establish strong conventions and sensible defaults
+
+- Use conventions to eliminate unnecessary decision-making and prevent bikeshedding. For example, require a `Makefile` with a `release-build` target in all applications rather than allow project teams to decide what command to use for building the container.
+- Adopt sensible defaults to minimize configuration overhead. For example, hardcode values (e.g., healthcheck timeout and interval) where flexibility is unnecessary.
+
+### Minimize the surface area of interfaces
+
+- Design interfaces with the minimum number of inputs required to meet the needs of most teams. For example, minimize the number of variables to Terraform modules and the number of inputs to GitHub Actions workflows.
+- Provide default values to inputs when possible.
+
+## Prefer configuration files over template variables
```

Comment:
This sections feels both too specific and to new to go into this document. Which is likely why it's so long. It might help to abstract it up a step?